pytest_lazy_fixtures/lazy_fixture_callable.py

```diff
@@ -1,30 +1,60 @@
 from __future__ import annotations
 
+import inspect
 from inspect import isfunction
 from typing import TYPE_CHECKING, Callable
 
-from .lazy_fixture import LazyFixtureWrapper
+from .lazy_fixture import LazyFixtureWrapper, lf
 
 if TYPE_CHECKING:
     import pytest
 
 
+def _fill_unbound_params(
+    func: Callable[..., object], args: tuple[object, ...], kwargs: dict[str, object]
+) -> dict[str, object]:
+    """Analyze a callable's signature and bind lazy fixtures to unbound parameters.
+
+    This function takes a callable and its provided arguments, examines the callable's
+    signature, and returns a dictionary that maps any unbound parameter names to their
+    corresponding lazy fixtures.
+
+    Only parameters that are not already bound through the provided args/kwargs will
+    get mapped to lazy fixtures. Any parameters that have existing values provided
+    maintain those values.
+    """
+    sig = inspect.signature(func)
+
+    bound = sig.bind_partial(*args, **kwargs)
+
+    # Apply the defaults arguments, as we don't want to override them.
+    bound.apply_defaults()
+
+    unbound = [name for name in sig.parameters if name not in bound.arguments]
+
+    return {unbound_param: lf(unbound_param) for unbound_param in unbound}
```

Comment:
You can use here LazyFixtureWrapper

```suggestion
    return {unbound_param: LazyFixtureWrapper(unbound_param) for unbound_param in unbound}
```